packages/framework-provider-local/src/services/read-model-registry.ts

```diff
@@ -44,7 +50,48 @@ export class ReadModelRegistry {
     if (limit) {
       cursor = cursor.limit(limit)
     }
-    return await cursor.execAsync()
+
+    const arrayFields: { [key: string]: any } = {}
+    select?.forEach((field: string) => {
+      const parts = field.split('.')
+      let currentLevel = arrayFields
+      let isArrayField = false
+      for (let i = 0; i < parts.length; ++i) {
+        const part = parts[i]
+        if (part.endsWith('[]')) {
+          const arrayField = part.slice(0, -2)
+          if (!Object.prototype.hasOwnProperty.call(currentLevel, arrayField)) {
+            currentLevel[arrayField] = {}
+          }
+          currentLevel = currentLevel[arrayField]
+          isArrayField = true
+        } else {
+          if (i === parts.length - 1) {
+            if (isArrayField) {
```

Comment:
A simplification here: 
```suggestion
        } else if (i === parts.length - 1 && isArrayField) {
```

Reply:
This suggestion broke the unit tests, so we'll ignore it for now.